azure-pipeline/src/index.ts

```diff
@@ -0,0 +1,113 @@
+import taskLib = require('azure-pipelines-task-lib/task');
+import { AxiosRequestConfig  } from "axios";
+import { updateUserConfig } from './utils/api';
+import {
+    API_VERSION,
+    SCOPE,
+    OUTPUT_PATH,
+    INPUT,
+} from './utils/constants';
+import {
+    FileHandler,
+} from './utils/file-handler';
+import {
+    validatedEnvVar,
+    validatedAuthParams,
+} from './utils/common';
+import { 
+    ClientSecretCredential,
+    AuthenticationError,
+} from '@azure/identity';
+import { info } from 'console';
+
+
+class ConfigUpdater {
+
+    private f = new FileHandler;
+
+    // getAuthorizationTokenFromKey():
+    // a function takes user's input given in the .yml file
+    // and request an bearer token for authentication of later authorization
+    // using @azure/identity
+    private getAuthorizationTokenFromKey = async () => {
+        try {
+            const params = {
+                tenantID:  validatedAuthParams('tenantid'),
```

Comment:
Reading a head, this `validatedAuthParams` function is getting values from the environment? I think a better, more understandable pattern that should be applied a cross this module would be to accept these values as parameters of the method. The caller can use `validatedAuthParams` to get the values or pass something else they know to be valid.

Reply:
Good point! I think make them params are good for flexibility and testability. I modified the function here as params and let the environment reading behavior sit outside when it is called.